dts/bindings/mtd/nxp,s32-qspi-hyperflash.yaml

```diff
@@ -0,0 +1,79 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  QSPI hyperflash connected to the NXP S32 QSPI bus.
+
+compatible: "nxp,s32-qspi-hyperflash"
+
+include: ["nxp,s32-qspi-device.yaml", "soc-nv-flash.yaml"]
+
+properties:
+  program-page-size:
+    type: int
+    default: 256
+    description: The programing page size in byte
+
+  device-id:
+    type: uint8-array
+    required: true
+    description: Device memory ID
```

Comment:
According to the DS of the hyperflash device connected on this board, it supports [SFDP JEDEC JESD216 serial flash discoverable parameters (SFDP) space](https://www.infineon.com/dgdl/Infineon-S26HS256T_S26HS512T_S26HS01GT_S26HL256T_S26HL512T_S26HL01GT_256Mb_512Mb_1Gb_SEMPER_TM_Flash_HYPERBUS_TM_interface_1-DataSheet-v67_00-EN.pdf?fileId=8ac78c8c7d0d8da4017d0ee6a0c96f54). So this binding should be inheriting from jesd216 as well

Reply:
I removed this property and updated to use "jedec-id"